ivy/functional/backends/paddle/activations.py

```diff
@@ -186,17 +172,17 @@ def log_softmax(
     return ret
 
 
+@with_supported_dtypes(
+    {"2.5.1 and below": ("float", "complex")},
+    backend_version,
+)
 def mish(x: paddle.Tensor, /, *, out: Optional[paddle.Tensor] = None) -> paddle.Tensor:
-    if x.dtype in unsupported_dtypes:
-        if paddle.is_complex(x):
-            return x * paddle_backend.tanh(paddle_backend.log1p(paddle_backend.exp(x)))
-        return F.mish(x.cast("float32")).cast(x.dtype)
+    if paddle.is_complex(x):
+        return x * paddle_backend.tanh(paddle_backend.log1p(paddle_backend.exp(x)))
     return F.mish(x)
 
 
-@with_unsupported_device_and_dtypes(
-    {"2.5.1 and below": {"cpu": ("float16",)}}, backend_version
-)
+@with_supported_dtypes({"2.5.1 and below": ("float",)}, backend_version)
```

Comment:
I think it should be based on whether our implementation works with those dtypes than the actual framework's implementation, because we might've added it due to tests failing or something which might not exactly be a native unsupported dtype error but some other error message 😄